src/main/java/blusunrize/immersiveengineering/common/util/network/MessageRequestBlockUpdate.java

```diff
@@ -32,25 +31,30 @@ public MessageRequestBlockUpdate()
 	@Override
 	public void fromBytes(ByteBuf buf)
 	{
-		dim = buf.readInt();
-		pos = new BlockPos(buf.readInt(), buf.readInt(), buf.readInt());
+		pos = BlockPos.fromLong(buf.readLong());
 	}
 
 	@Override
 	public void toBytes(ByteBuf buf)
 	{
-		buf.writeInt(dim).writeInt(pos.getX()).writeInt(pos.getY()).writeInt(pos.getZ());
+	    buf.writeLong(pos.toLong());
 	}
 
 	public static class Handler implements IMessageHandler<MessageRequestBlockUpdate, IMessage>
 	{
 		@Override
 		public IMessage onMessage(MessageRequestBlockUpdate message, MessageContext ctx)
 		{
-			synchronized (EventHandler.requestedBlockUpdates)
-			{
-				EventHandler.requestedBlockUpdates.offer(new ImmutablePair<Integer, BlockPos>(message.dim, message.pos));
-			}
+			WorldServer world = ctx.getServerHandler().player.getServerWorld();
+			world.addScheduledTask(() -> {
+				synchronized (EventHandler.requestedBlockUpdates)
+				{
+					if (world.isBlockLoaded(message.pos)) {
+						int dim = world.provider.getDimension();
+						EventHandler.requestedBlockUpdates.offer(new ImmutablePair<>(dim, message.pos));
```

Comment:
Alternatively consider using a java.util.concurrent Queue, so callers need not worry about synchronized().